File: App/Resources/AlgorithmDetails/bubblebogosort/bubblebogosort.go
package main

import "fmt"

func sort(a []int) []int {
	n := len(a)
	if n < 2 {
		return a
	}
	swapped := true
	for swapped {
		swapped = false
		for i := 0; i+1 < n; i++ {
			if a[i] > a[i+1] {
				a[i], a[i+1] = a[i+1], a[i]
				swapped = true
			}
		}
	}
	return a
}
func main() { array := []int{0, 39, 21, 62, 91, 77, 14, 23}; fmt.Println(sort(array)) }

File: App/Resources/AlgorithmDetails/bubblebogosort/bubblebogosort.swift
func sort(_ a: inout [Int]) {
    let n = a.count
    guard n > 1 else { return }
    var swapped = true
    while swapped {
        swapped = false
        for i in 0 ..< (n - 1) where a[i] > a[i + 1] {
            a.swapAt(i, i + 1)
            swapped = true
        }
    }
}

var array = [0, 39, 21, 62, 91, 77, 14, 23]
sort(&array)
print(array)
